samples/previews/azure-functions/service-bus/ASBFunctions_0/AzureFunctions.ASBTrigger.FunctionsHostBuilder/Startup.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using NServiceBus;
+
+#region configuration-with-function-host-builder
+
+[assembly: FunctionsStartup(typeof(Startup))]
+
+public class Startup : FunctionsStartup
+{
+    public override void Configure(IFunctionsHostBuilder builder)
+    {
+        var services = builder.Services;
+
+        // register custom service in the container
+        services.AddSingleton(_ =>
+        {
+            var configurationRoot = builder.GetContext().Configuration;
+            var customComponentInitializationValue = configurationRoot.GetValue<string>("CustomComponentValue");
+
+            return new CustomComponent(customComponentInitializationValue);
+        });
+
+        builder.UseNServiceBus(() =>
+        {
+            var configuration = new ServiceBusTriggeredEndpointConfiguration(AzureServiceBusTriggerFunction.EndpointName);
+
+            // optional: log startup diagnostics using Functions provided logger
```

Comment:
not sure I understand this comment? The Logging will use the functions provided logger?

Reply:
We keep it as-is to show that logging diagnostics is possible and it's optional.